lib/redis/cluster.rb

```diff
@@ -69,7 +69,12 @@ def with_reconnect(val = true, &block)
     end
 
     def call(command, &block)
-      send_command(command, &block)
+      begin
+        send_command(command, &block)
+      rescue CannotConnectError
+        update_cluster_info!
```

Comment:
Should we reject the `fail` node in the following code?

https://github.com/redis/redis-rb/blob/78bf94051d8176a1a5721b8647a1ceb730119ab2/lib/redis/cluster/node_loader.rb#L24-L32

```
$ make start_cluster
$ make create_cluster
$ telnet localhost 7000
Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.
shutdown
Connection closed by foreign host.

$ telnet localhost 7001
Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.
cluster nodes
$728
e261a9204067566d27f3b97b1418017ce65171f3 127.0.0.1:7001@17001 myself,master - 0 1568692342000 2 connected 5461-10922
f4093471beaad607406d1933d6e0eb7c7bdb4d90 127.0.0.1:7000@17000 master,fail - 1568692271534 1568692270531 1 disconnected
8613ac5796f3516b2eee2321cb82a78648fe0eb1 127.0.0.1:7003@17003 master - 0 1568692343143 10 connected 0-5460
0de04c346cf9c5b24a076e80073f6bfc4377a866 127.0.0.1:7004@17004 slave e261a9204067566d27f3b97b1418017ce65171f3 0 1568692344148 2 connected
0a1054fbb6504d392989a7492efbaf5731ee65cc 127.0.0.1:7002@17002 master - 0 1568692343545 3 connected 10923-16383
36352f5a62cb950e4d1b26c28adf5ddc47be7fe7 127.0.0.1:7005@17005 slave 0a1054fbb6504d392989a7492efbaf5731ee65cc 0 1568692343000 6 connected
```

In the above case, the `127.0.0.1:7000` node is failing.

Reply:
hmm, the failed node may rejoin as a slave. Moreover, since the call is done for each node, it shouldn't fail even if we don't remove the failed node
```go
        nodes.each do |node|
          info = fetch_node_info(node)
          info.empty? ? next : break
        end
```

Reply:
I see. :+1: